### PR TITLE
feat: port Algebra.Order.Hom.Basic again

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
---

Partially supersedes https://github.com/leanprover-community/mathlib4/pull/1238.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by eric-wieser** [comment]
Marking WIP in light of [this Zulip thread](https://leanprover.zulipchat.com/#narrow/stream/144837-PR-reviews/topic/!4.231238/near/350169875).

File: Mathlib/Algebra/Order/Hom/Basic.lean [hunk 1]
```diff
@@ -4,13 +4,11 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Yaël Dillies
 
 ! This file was ported from Lean 3 source module algebra.order.hom.basic
-! leanprover-community/mathlib commit 28aa996fc6fb4317f0083c4e6daf79878d81be33
+! leanprover-community/mathlib commit 7ea604785a41a0681eac70c5a82372493dbefc68
```
Comment (eric-wieser) [inline Mathlib/Algebra/Order/Hom/Basic.lean:7]: The [port-status page](https://leanprover-community.github.io/mathlib-port-status/file/algebra/order/hom/basic?range=28aa996fc6fb4317f0083c4e6daf79878d81be33..7ea604785a41a0681eac70c5a82372493dbefc68) is telling me that this change is a no-op; are you sure this file actually needs re-porting?

File: Mathlib/Algebra/Order/Hom/Basic.lean [hunk 2]
```diff
@@ -139,7 +137,6 @@ theorem le_map_add_map_div [Group α] [AddCommSemigroup β] [LE β] [MulLEAddHom
     (a b : α) : f a ≤ f b + f (a / b) := by
   simpa only [add_comm, div_mul_cancel'] using map_mul_le_add f (a / b) b
 #align le_map_add_map_div le_map_add_map_div
--- #align le_map_add_map_sub le_map_add_map_sub -- Porting note: TODO: `to_additive` clashes
```
Comment (eric-wieser) [inline Mathlib/Algebra/Order/Hom/Basic.lean:142]: Why was this removed?

**Review by eric-wieser — COMMENTED** [review]
I don't really understand this PR; the file is already in sync, and the changes seem to only be minor syntactic tweaks, and deletion of porting notes without comment.

**Comment by eric-wieser** [comment]
Perhaps @Parcly-Taxel's e96eb2efeda7f5b43236bf0d9093791c4ca21456 commit is causing the confusion

File: Mathlib/Algebra/Order/Hom/Basic.lean (hunk 1 above)
Comment (Parcly-Taxel) [reply Mathlib/Algebra/Order/Hom/Basic.lean:7]: The port status page is wrong on this file. The most recent commit that modified this file is https://github.com/leanprover-community/mathlib/commit/7ea604785a41a0681eac70c5a82372493dbefc68, whose SHA does not match the SHA of the file in mathlib4.

Comment (Parcly-Taxel) [reply Mathlib/Algebra/Order/Hom/Basic.lean:7]: The claimed "[last sync](https://github.com/leanprover-community/mathlib/commit/28aa996fc6fb4317f0083c4e6daf79878d81be33)" on the port status page does not touch `Algebra.Order.Hom.Basic` at all.

File: Mathlib/Algebra/Order/Hom/Basic.lean (hunk 2 above)
Comment (Parcly-Taxel) [reply Mathlib/Algebra/Order/Hom/Basic.lean:142]: Because the current mathport output for the file has that exact same line in the exact same place.

File: Mathlib/Algebra/Order/Hom/Basic.lean (hunk 1 above)
Comment (eric-wieser) [reply Mathlib/Algebra/Order/Hom/Basic.lean:7]: What's happened here is that we changed our SHA-labelling strategy from "any commit in mathlib3" to "only commits that modifies the file in question in mathlib3".

There are many SHAs that are imprecise in this way, but it is harmless and the out of sync page does not care.

Comment (Parcly-Taxel) [reply Mathlib/Algebra/Order/Hom/Basic.lean:7]: Still, there are substantative differences between what is currently in mathlib4 and this PR (i.e. the cleaned up current mathport output). The former imports two other files, the latter only one, for example.

File: Mathlib/Algebra/Order/Hom/Basic.lean (hunk 2 above)
Comment (eric-wieser) [reply Mathlib/Algebra/Order/Hom/Basic.lean:142]: I don't understand what you mean, can you elaborate?

File: Mathlib/Algebra/Order/Hom/Basic.lean [hunk 3]
```diff
@@ -151,21 +148,8 @@ theorem le_map_div_mul_map_div [Group α] [CommSemigroup β] [LE β] [Submultipl
...
-
---Porting note: tactic extension commented as decided in the weekly porting meeting
--- /-- Extension for the `positivity` tactic: nonnegative maps take nonnegative values. -/
--- @[positivity _ _]
--- unsafe def positivity_map : expr → tactic strictness
---   | expr.app (quote.1 ⇑(%%f)) (quote.1 (%%ₓa)) => nonnegative <$> mk_app `` map_nonneg [f, a]
---   | _ => failed
--- #align tactic.positivity_map tactic.positivity_map
-
---end Mathlib.Meta.Positivity
```
Comment (eric-wieser) [inline Mathlib/Algebra/Order/Hom/Basic.lean:168]: What was the reason for deleting this?

File: Mathlib/Algebra/Order/Hom/Basic.lean (hunk 2 above)
Comment (Parcly-Taxel) [reply Mathlib/Algebra/Order/Hom/Basic.lean:142]: With that line uncommented I get this error message:
`le_map_add_map_sub has already been aligned (to le_map_add_map_sub)`
Since `to_additive` has been improved to the point where it can take care of this all by itself, I feel this doesn't warrant a porting note anymore.

File: Mathlib/Algebra/Order/Hom/Basic.lean (hunk 3 above)
Comment (Parcly-Taxel) [reply Mathlib/Algebra/Order/Hom/Basic.lean:168]: This meta code doesn't appear at all in the latest mathport output:
https://github.com/leanprover-community/mathlib3port/blob/master/Mathbin/Algebra/Order/Hom/Basic.lean#L186-L195

File: Mathlib/Algebra/Order/Hom/Basic.lean (hunk 2 above)
Comment (eric-wieser) [reply Mathlib/Algebra/Order/Hom/Basic.lean:142]: Ah nevermind, I understand what's happening here now (and it's not what you described).

Because this is `to_additive existing`, there is a line in another file that is manually doing the correct alignment.

File: Mathlib/Algebra/Order/Hom/Basic.lean [hunk 4]
```diff
@@ -222,11 +206,11 @@ export AddGroupNormClass (eq_zero_of_map_eq_zero)
 
 export GroupNormClass (eq_one_of_map_eq_zero)
 
-attribute [simp] map_one_eq_zero -- porting note: `to_additive` translation already exists
```
Comment (eric-wieser) [inline Mathlib/Algebra/Order/Hom/Basic.lean:225]: What's the story behind this removal?

Comment (Parcly-Taxel) [reply Mathlib/Algebra/Order/Hom/Basic.lean:225]: In mathport (https://github.com/leanprover-community/mathlib3port/blob/43c332dc22f5a04dfba2d10ea68e73f56a5df063/Mathbin/Algebra/Order/Hom/Basic.lean#L251) this line is
`attribute [simp, to_additive map_zero] map_one_eq_zero`
However this line with `to_additive` gives the following errors:
```
Basic.lean:210:29
The additive declaration already exists. Please specify this explicitly using `@[to_additive existing]`. [linter.toAdditiveExisting]
Basic.lean:210:17
The translation GroupSeminormClass.map_one_eq_zero ↦ AddGroupSeminormClass.map_zero already exists
```
So `to_additive map_zero` must be removed.

Comment (eric-wieser) [reply Mathlib/Algebra/Order/Hom/Basic.lean:225]: Does `attribute [simp, to_additive existing map_zero] map_one_eq_zero` still fail?

Comment (Parcly-Taxel) [reply Mathlib/Algebra/Order/Hom/Basic.lean:225]: Yes, it still fails, and just with the `The translation GroupSeminormClass.map_one_eq_zero ↦ AddGroupSeminormClass.map_zero already exists` error message.

Comment (eric-wieser) [reply Mathlib/Algebra/Order/Hom/Basic.lean:225]: I think we should keep the porting note: it flags that this is a mistake in mathlib3 that we were forced to fix during porting

Comment (Parcly-Taxel) [reply Mathlib/Algebra/Order/Hom/Basic.lean]: Right then.

File: Mathlib/Algebra/Order/Hom/Basic.lean (hunk 1 above)
Comment (Parcly-Taxel) [reply Mathlib/Algebra/Order/Hom/Basic.lean:7]: @eric-wieser Is there anything else you want to say?

Comment (eric-wieser) [reply Mathlib/Algebra/Order/Hom/Basic.lean:7]: I think "substantive" is rather a strong interpretation, but indeed we should correct the out-of-sync imports.

Comment (Parcly-Taxel) [reply Mathlib/Algebra/Order/Hom/Basic.lean:7]: And the SHA? Is this PR ready to merge as-is?

Comment (eric-wieser) [reply Mathlib/Algebra/Order/Hom/Basic.lean:7]: Changing the SHA is probably not a good idea; in general moving SHAs backwards is going to make things harder for tools.

EIther way, I'd like to hear what @Ruben-VandeVelde intended to do with this PR, since it seems like they thought there was more to forward-port than a change of import.

**Comment by jcommelin** [comment]
Marking this as `awaiting-author`, since it would be good to hear @Ruben-VandeVelde's input.
